apis/v1alpha2/validation/gateway_test.go

```diff
@@ -0,0 +1,162 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package validation
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	gatewayv1a2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+func TestValidateGateway(t *testing.T) {
+	listeners := []gatewayv1a2.Listener{
+		{
+			Hostname: nil,
+		},
+	}
+	baseGateway := gatewayv1a2.Gateway{
```

Comment:
```golang
baseGateway := gatewayv1a2.Gateway{
	ObjectMeta: metav1.ObjectMeta{
			Name:      "foo",
			Namespace: metav1.NamespaceDefault,
		},
		Spec: gatewayv1a2.GatewaySpec{
			GatewayClassName: "foo",
			Listeners:        []gatewayv1a2.Listener{
		            {
			      Hostname: nil,
		            },
		},
	}
```
instead of two steps object construction.